go/enclave/enclave.go

```diff
@@ -561,6 +562,37 @@ func (e *enclaveImpl) Stop() error {
 	return nil
 }
 
+// EstimateGas decrypts CallMsg data, runs the gas estimation for the data.
+// Using the callMsg.From Viewing Key, returns the encrypted gas estimation
+func (e *enclaveImpl) EstimateGas(encryptedParams common.EncryptedParamsEstimateGas) (common.EncryptedResponseEstimateGas, error) {
+	// decrypt the input with the enclave PK
+	paramBytes, err := e.rpcEncryptionManager.DecryptBytes(encryptedParams)
+	if err != nil {
+		return nil, fmt.Errorf("unable to decrypt params in EstimateGas request. Cause: %w", err)
+	}
+
+	// extract params from byte slice to array of strings
+	var paramString []string
+	err = json.Unmarshal(paramBytes, &paramString)
+	if err != nil {
+		return nil, fmt.Errorf("unable to decode EstimateGas params - %w", err)
+	}
+
+	// marshall the string into a callMsg
+	callMsg, err := rpc.ExtractCallMsg(gethcommon.Hex2Bytes(paramString[0][2:]))
```

Comment:
@otherview Can you add a comment explaining the slicing here?

Reply:
Also added some more testing 👍